src/components/ResourceList/ResourceList.tsx

```diff
@@ -611,39 +344,40 @@ class ResourceListInner extends React.Component<CombinedProps, State> {
       const overflow = viewportHeight - overlay.height;
 
       const spinnerHeight =
-        this.props.items.length === 1
-          ? SMALL_SPINNER_HEIGHT
-          : LARGE_SPINNER_HEIGHT;
+        items.length === 1 ? SMALL_SPINNER_HEIGHT : LARGE_SPINNER_HEIGHT;
 
       const spinnerPosition =
         overflow > 0
           ? (overlay.height - spinnerHeight) / 2
           : (viewportHeight - overlay.top - spinnerHeight) / 2;
 
-      this.setState({loadingPosition: spinnerPosition});
+      setLoadingPositionState(spinnerPosition);
     }
-  };
+  }, [listRef, items.length]);
 
-  private handleSelectAllItemsInStore = () => {
-    const {
-      onSelectionChange,
-      selectedItems,
-      items,
-      idForItem = defaultIdForItem,
-    } = this.props;
+  const itemsExist = useCallback(
+    (itemsParam?: ItemType[]) => {
+      return (itemsParam || items).length > 0;
+    },
+    [items],
+  );
```

Comment:
```suggestion
const itemsExist = (itemsParam || items).length > 0
```

We have no need for this to be a function anymore. iirc previously it was a `getter` than because of a typescript bump we had to change that to regular functional making things a little more confusing.

Reply:
Good call. The useCallback was really throwing me for a loop. I committed the boolean change.